chirp/directory.py

```diff
@@ -120,6 +120,16 @@ def get_driver(rclass):
 
 def get_radio_by_image(image_file):
     """Attempt to get the radio class that owns @image_file"""
+    # Creating temp file for csv comment support
+    if image_file.endswith(".csv"):
+        with open(image_file, 'r') as origfile, \
+                open(image_file + '.chirptmp.csv', 'w') as tempfile:
+            for line in origfile:
+                # exclude lines starting with comment character "#"
+                if not line.startswith('#'):
+                    tempfile.write(line)
+        image_file = image_file + '.chirptmp.csv'
+
```

Comment:
This is really not a good way to do this. For one, it doesn't catch files that end in `.CSV` but it also may try to convert files that claim to be CSV in name but don't otherwise contain that kind of content. We really just need to filter the content as it is read. I can help with that if necessary, and I'd definitely appreciate the functionality of being able to tolerate comments. We just need to remember that they'll be stripped out with an open-save cycle, unless we do something more complicated.

Reply:
Agreed, not a great way to do it, and I was hoping to do it while reading the file (i.e.  I was hoping for a single read loop or something, where I could add a line that filters out comments), but it looks like .csv files are read two different ways in Chirp, once with a binary read into filedata in directory.py and then using csv.reader in generic_csv.py, so I used the temp file option to just do it in one place in the code and make it work.  After thinking on your comments more, I dug a little deeper and now have it filtering out the lines in two different places, but without having to use a temp file.

As for catching files that end in .csv but which aren't .csv files, I'm not sure what to do about that.  If someone calls a file a .csv file, but it isn't, then it seems like they're trying to do something wrong and might get whatever results, they get, no?  And isn't that the same behavior they'd get right now?   Do you currently have code that verifies that a file opened with a .csv extension actually is a .csv file before doing anything with it?  If so, let me know where, and maybe I could reuse that code.

Anyway, I rewrote the code to not use temp files and will update my branch in my fork with the new code (I think that's what I should do.  If it looks like that didn't work, I'll close this request and open a new one with my updated branch...  Let me know what you think.  :-)

Also, for future reference, if there is an easier way to "update a pull request" with new code, let me know.